taskvine/src/worker/vine_cache.c

```diff
@@ -376,99 +374,160 @@ static int do_transfer( struct vine_cache *c, const char *source_url, const char
 	return result;
 }
 
+/* 
+scan through transfer processes and mark them as completed as they finish.
+*/
+
+int vine_cache_wait(struct vine_cache *c, struct link *manager)
+{
+	int status;
+	int exit_code;
+	struct cache_file *f;
+    	char * cachename;
+	HASH_TABLE_ITERATE(c->table, cachename, f){
+		if(f->pid && f->status==VINE_FILE_STATUS_PROCESSING){
+			int result = waitpid(f->pid, &status, WNOHANG);
+			if(result !=0 && f->type==VINE_CACHE_MINI_TASK){
```

Comment:
This seems like a strange place to put this, shouldn't it be part of the `result>0` clause below with the rest of the success case?